fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/ConsensusNode.java

```diff
@@ -36,15 +36,15 @@ public String getPublicKey() {
     return Collections.unmodifiableSet(acceptedMessageIds);
   }
 
-  public Optional<Consensus> getLastConsensus(String keyId) {
-    // get the consensus for the given key id, with the largest creation time
+  public Optional<Consensus> getLastConsensus(String instanceId) {
+    // get the consensus for the given instanceId, with the largest creation time
     return consensuses.stream()
-        .filter(consensus -> consensus.getKey().getId().equals(keyId))
+        .filter(consensus -> consensus.getId().equals(instanceId))
         .max((c1, c2) -> (int) (c1.getCreation() - c2.getCreation()));
```

Comment:
I feel like it would be easier to read like this

```suggestion
        .max(Comparator.comparingLong(Consensus::getCreation));
```

You need to import `java.util.Comparator`